source/common/http/utility.cc

```diff
@@ -29,8 +29,145 @@
 #include "absl/strings/str_cat.h"
 #include "absl/strings/str_split.h"
 #include "absl/strings/string_view.h"
+#include "nghttp2/nghttp2.h"
 
 namespace Envoy {
+namespace Http2 {
+namespace Utility {
+
+namespace {
+
+void validateCustomSettingsParameters(
```

Comment:
OK I see here that we are doing the validation. Can we then remove the warnings in the datapath since it shouldn't ever happen?

Reply:
Done. I kept a warning for the exception (`allow_connect`) as I think it's helpful to notify end users of bad configs when possible.